pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1649,7 +1649,13 @@ func PodTopologySpreadPredicate(pod *v1.Pod, meta *PodTopologySpreadMetadata, no
 		return false, nil, fmt.Errorf("node not found")
 	}
 
-	if meta == nil || len(meta.tpPairToMatchNum) == 0 || len(meta.constraints) == 0 {
+	// nil meta is illegal.
+	if meta == nil {
+		return false, nil, errors.New("metadata not pre-computed for PodTopologySpreadPredicate")
```

Comment:
We need a todo here to implement this, right?

Reply:
done.